src/devices/Media/VideoDevice/Resolution.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System;
+
+namespace Iot.Device.Media
+{
+    /// <summary>
+    /// The resolution type of a video device.
+    /// </summary>
+    public enum ResolutionType : uint
+    {
+        /// <summary>
+        /// Discrete
+        /// </summary>
+        Discrete = v4l2_frmsizetypes.V4L2_FRMSIZE_TYPE_DISCRETE,
+
+        /// <summary>
+        /// Continuous
+        /// </summary>
+        Continuous = v4l2_frmsizetypes.V4L2_FRMSIZE_TYPE_CONTINUOUS,
+
+        /// <summary>
+        /// Stepwise
+        /// </summary>
+        Stepwise = v4l2_frmsizetypes.V4L2_FRMSIZE_TYPE_STEPWISE
+    }
+
+    /// <summary>
+    /// Resolution's interface
+    /// </summary>
+    public interface IResolution
+    {
+        /// <summary>
+        /// Resolution's type
+        /// </summary>
+        ResolutionType Type { get; }
```

Comment:
the type is also in the C# type itself, you can do:

`if (myIResolution is DiscreteResolution)`. Would it make sense to make this class have some common utilities? i.e. MinWidth, MaxWidth, MinHeight, MaxHeight? For DiscreteResolution that would pairwise return the same value but still would make it much easier to use

Reply:
take it with the grain of salt, I'm not super familiar with this code, I just like easy user experience if possible 😄

Reply:
It's possible to implement it as a single class using Min and Max, with StepHeight and StepWidth  values of 0, 1, n based on the type. Only discrete resolution will not be matching the v4l2 implementation, but I suppose it should be fine.